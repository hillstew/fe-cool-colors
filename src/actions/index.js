export const setColors = colors => ({
  type: 'SET_COLORS',
  colors
});

export const setLoading = bool => ({
  type: 'SET_LOADING',
  bool
});

export const setError = error => ({
  type: 'SET_ERROR',
  error
});

export const setProjects = projects => ({
  type: 'SET_PROJECTS',
  projects
});

export const setPalettes = palettes => ({
  type: 'SET_PALETTES',
  palettes
});

export const toggleLocked = id => ({
  type: 'TOGGLE_LOCKED',
  id
});
