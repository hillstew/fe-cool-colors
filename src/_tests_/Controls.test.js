import React from 'react';
import { shallow } from 'enzyme';
import {
  Controls,
  mapStateToProps,
  mapDispatchToProps
} from '../containers/Controls/Controls';
import { toggleLocked, setColors } from '../actions';

describe('Controls', () => {
  describe('Controls component', () => {
    let wrapper;
    const mockProps = {
      postProject: jest.fn(() => true),
      projects: [
        {
          name: 'Movie Tracker',
          id: 'abcdefg'
        }
      ],
      colors: ['red', 'orange', 'yellow', 'green', 'blue'],
      setColors: jest.fn()
    };

    beforeEach(() => {
      wrapper = shallow(<Controls {...mockProps} />);
    });

    it('should match the snapshot', () => {
      expect(wrapper).toMatchSnapshot();
    });
  });

  describe('mapStateToProps', () => {});

  describe('mapDispatchToProps', () => {});
});
