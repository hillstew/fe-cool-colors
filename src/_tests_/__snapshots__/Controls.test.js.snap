// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Controls Controls component should match the snapshot 1`] = `
<div
  className="color-controls"
>
  <button
    className="generate-color-button"
    onClick={[Function]}
  >
    Generate!
  </button>
  <select
    name="projectSelection"
    onChange={[Function]}
    value=""
  >
    <option
      key="abcdefg"
      value="Movie Tracker"
    >
      Movie Tracker
    </option>
    <option
      value="New Project"
    >
      New Project
    </option>
  </select>
  <button
    onClick={[Function]}
  >
    Save
  </button>
</div>
`;
